adcc/backends/test_pyscf.py

```diff
@@ -159,7 +152,22 @@ def template_rhf_h2o(self, basis):
         ao_dip = mf.mol.intor_symmetric('int1e_r', comp=3)
         operator_import_test(mf, list(ao_dip))
 
+    def template_uhf_h2o(self, basis):
+        mf = adcc.backends.run_hf(
+            "pyscf", geometry.xyz["h2o"], basis, multiplicity=3
+        )
+        self.base_test(mf)
+
+        # Test ERI
+        eri_asymm_construction_test(mf)
+        eri_asymm_construction_test(mf, core_orbitals=1)
+
+        # Test dipole
+        ao_dip = mf.mol.intor_symmetric('int1e_r', comp=3)
+        operator_import_test(mf, list(ao_dip))
+
     def test_h2o_sto3g_core_hole(self):
+        pytest.xfail("Systems with a core-hole are currently not supported.")
```

Comment:
I'm not happy about that, because that's a feature we do want and need. Where does it fail?

Reply:
Actually the example still seems to work and produce the right results.

Reply:
I think we should add a "proper" core-hole test including an XES calculation and comparison of energies ... I'll do that.

Reply:
XES test passes now ... the point is that with the new interface and mechanisms, the requirements that the occupied orbitals are sorted first is no longer given, since the MoSpaces now resorts as needed. I'll proceed to add a functionality test for XES.

Reply:
I really don't know why this works... but if you do, I'm okay with it 😄

Reply:
The reordering py-side was a hack because the "old" adcc-MoSpaces-interface did assume the first mos were occupied and the later ones were not. This has been changed a while back. Essentially the fact that core-hole states have an empty core orbital now no longer matters as far as the adcc interface is concerned, so we do not need to assert and test this. That's all there is to it.